pkg/bloomcompactor/v2_meta.go

```diff
@@ -0,0 +1,134 @@
+package bloomcompactor
+
+import (
+	"fmt"
+	"hash"
+	"path"
+
+	v1 "github.com/grafana/loki/pkg/storage/bloom/v1"
+	"github.com/grafana/loki/pkg/storage/stores/shipper/indexshipper/tsdb"
+	"github.com/grafana/loki/pkg/util/encoding"
+	"github.com/pkg/errors"
+)
+
+const (
+	BloomPrefix = "bloom"
+)
+
+// TODO(owen-d): Probably want to integrate against the block shipper
+// instead of defining here, but only (min,max,fp) should be required for
+// the ref. Things like index-paths, etc are not needed and possibly harmful
+// in the case we want to do migrations. It's easier to load a block-ref or similar
+// within the context of a specific tenant+period+index path and not couple them.
+type BlockRef struct {
+	OwnershipRange v1.FingerprintBounds
+	Checksum       uint32
+}
+
+func (r BlockRef) Hash(h hash.Hash32) error {
+	if err := r.OwnershipRange.Hash(h); err != nil {
+		return err
+	}
+
+	var enc encoding.Encbuf
+	enc.PutBE32(r.Checksum)
+	_, err := h.Write(enc.Get())
+	return errors.Wrap(err, "writing BlockRef")
+}
+
+type MetaRef struct {
+	OwnershipRange v1.FingerprintBounds
+	Checksum       uint32
+}
+
+// `bloom/<period>/<tenant>/metas/<start_fp>-<end_fp>-<start_ts>-<end_ts>-<checksum>.json`
+func (m MetaRef) Address(tenant string, period int) (string, error) {
+	joined := path.Join(
+		BloomPrefix,
+		fmt.Sprintf("%v", period),
+		tenant,
+		"metas",
+		m.OwnershipRange.String(),
+		fmt.Sprintf("%v", m.Checksum),
```

Comment:
`<start>-<end>-<checksum>`

```suggestion
		strings.Join([]string{m.OwnershipRange.String(), fmt.Sprintf("%x", m.Checksum)}, "-")
```